woodwork/table_accessor.py

```diff
@@ -206,8 +265,9 @@ def _check_logical_types(dataframe_columns, logical_types):
 def _check_schema(dataframe, schema):
     if not isinstance(schema, Schema):
         raise TypeError('Provided schema must be a Woodwork.Schema object.')
-    if not _is_valid_schema(dataframe, schema):
-        raise ValueError('Invalid typing information presented at init. Cannot set Woodwork on DataFrame.')
+    invalid_schema_message = _get_invalid_schema_message(dataframe, schema)
+    if invalid_schema_message:
+        raise ValueError(f'Woodwork typing information is not valid for this DataFrame: {invalid_schema_message}')
```

Comment:
@thehomebrewnerd The one place I've allowed to error for an invalid schema is when you directly initialize with a Schema (so calling `df.ww.init(schema)` directly instead of going through `df.ww.sample` or something).

The three situations where Woodwork will check for an invalid schema are as follows:

- Perform a pandas action that creates a new DataFrame for which the schema is invalid - WARNS
- Perform a pandas action that alters the DataFrame in a way that invalidates the schema - WARNS
- Directly initialize - ERRORS

My logic is that I think we want to be allowed to raise errors when users call `df.ww.init` directly. If we say that `df.ww.init(schema)` warns instead of errors, then I think it’d follow that we should also be warning users if they are setting individual parameters with `df.ww.init(logical_types={..}, index='id')`, but I think we definitely want to raise an Error if you try to set the index to an invalid column.